common/lanelet2_extension/lib/CarmaTrafficLight.cpp

```diff
@@ -75,21 +75,16 @@ boost::optional<CarmaTrafficLightState> CarmaTrafficLight::predictState(ros::Tim
   ros::Duration accumulated_offset_duration;
   double offset_duration_dir = recorded_time_stamps.front().first > time_stamp ? -1.0 : 1.0; // -1 if past, +1 if time_stamp is in future
 
+  int num_of_cycles = std::abs((recorded_time_stamps.front().first - time_stamp).toSec()/ fixed_cycle_duration.toSec());
+  accumulated_offset_duration = ros::Duration( num_of_cycles * fixed_cycle_duration.toSec());
+  
```

Comment:
I'm not convinced by this calculation. Take the following case
G end = 1000
Y end = 2000
R end = 3000
time_stamp = 500 (so it should be in the green phase)
fixed_cycle_duration = 3000 (since red ends at 3000)

num_of_cycles = int((1000 - 500 ) / (3000)) = 0
accumulated_offset_duration = 0 * 3000 = 0

offset_duration_dir  = -1 ( since 1000 > 5000)

so we will enter the while loop below for one iteration
after the loop accumulated_offset_duration = 0 + 3000 = 3000

Then we enter the for loop below and get 
1000 + -1 * 3000 = -2000
-2000 >= 500 => false
Loop will spin infinitely

Reply:
As discussed on slack, please add exception and log after for loop in the end for case where the duplicate phase is not provided.